src/SOS/Strike/strike.cpp

```diff
@@ -9202,7 +9212,42 @@ DECLARE_API(u)
     }
 
     NameForMD_s(methodDesc, g_mdName, mdNameLen);
-    ExtOut("%S\n", g_mdName);   
+    ExtOut("%S\n", g_mdName);
+
+    if (bIL)
+    {
+        ToRelease<IXCLRDataMethodInstance> pMethodInst(NULL);
+
+        if ((Status = GetClrMethodInstance(codeHeaderData.MethodStart, &pMethodInst)) != S_OK)
+        {
+            return Status;
+        }
+
+        CLRDATA_IL_ADDRESS_MAP* map = nullptr;
+        ULONG32 mapCount = 0;
+
+        if ((Status = pMethodInst->GetILAddressMap(mapCount, &mapCount, map)) != S_OK)
+        {
+            return Status;
+        }
+
+        map = new CLRDATA_IL_ADDRESS_MAP[mapCount];
+
+        if ((Status = pMethodInst->GetILAddressMap(mapCount, &mapCount, map)) != S_OK)
+        {
+            return Status;
```

Comment:
@cshung this look like a memory leak to me because you never deleted the map when this happens. Am I reading this correctly because I stumbled upon this PR. FTR I don't have a clue about this project :)

Reply:
It will be better to use a STL vector instead of manually new/delete.

```c++

        ULONG32 mapCount = 0;

        if ((Status = pMethodInst->GetILAddressMap(mapCount, &mapCount, nullptr)) != S_OK)
        {
            return Status;
        }

        // this is too risky
        // map = new CLRDATA_IL_ADDRESS_MAP[mapCount];
        std::vector<CLRDATA_IL_ADDRESS_MAP> maps(mapCount);

        if ((Status = pMethodInst->GetILAddressMap(maps.size(), &mapCount, &maps[0])) != S_OK)
        {
            return Status;
        }

        for (ULONG32 i = 0; i < mapCount; i++)
        {
            // TODO: These information should be interleaved with the disassembly
            // Decoded IL can be obtained through refactoring DumpIL code.
            ExtOut("%04x %p %p\n", maps[i].ilOffset, maps[i].startAddress, maps[i].endAddress);
        }
        // don't need this now
        // delete[] map;
    }
```